src/Control/Effect/Reader.hs

```diff
@@ -56,9 +56,9 @@ newtype ReaderC r m a = ReaderC { runReaderC :: r -> m a }
 
 instance (Carrier sig m, Monad m) => Carrier (Reader r :+: sig) (ReaderC r m) where
   ret a = ReaderC (const (ret a))
-  eff op = ReaderC (\ r -> (alg r \/ eff . handleReader r runReaderC) op)
-    where alg r (Ask       k) = runReaderC (k r) r
-          alg r (Local f m k) = runReaderC m (f r) >>= flip runReaderC r . k
+  eff op = ReaderC (\ r -> handleSum (eff . handleReader r runReaderC) (\case
+    Ask       k -> runReaderC (k r) r
+    Local f m k -> runReaderC m (f r) >>= flip runReaderC r . k) op)
```

Comment:
Unfortunately we still need the parentheses around the `LambdaCase`, because of the `op` parameter following it.